### PR TITLE
Add testing template

### PR DESCRIPTION
**Pull request by MichaelClifford**
This PR adds a testing template we can use to build our e2e tests for the distributed workloads/CodeFlare Stack.

* The template includes an empty function for each of the test issues we outlined on our [issues](https://github.com/opendatahub-io/distributed-workloads/issues) page. But feel free to supplement with more supporting functions if needed. 

* I added a small subset of tests for the KubeRay operator to get us started and provide an example of how to write tests in this format. 

* There is also a `resources` directory where we can add whatever scripts, yaml files, etc we will need during testing. It is currently just populated with template python file.

### CODE REVIEW

File: tests/basictests/distributed-workloads.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+install_codeflare_operator
+install_distributred_workloads_kfdef
+test_mcad_torchx_functionality
+tests_mcad_ray_functionality
+uninstall_distributed_workloads_kfdef
+uninstall_codeflare_operator
+
+
+os::test::junit::declare_suite_end
```
Comment (Maxusmusti) [inline tests/basictests/distributed-workloads.sh:57]: missing newline at EOF

File: tests/resources/test_mcad_ray.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Import pieces from codeflare_sdk
+from codeflare_sdk.cluster.cluster import Cluster, ClusterConfiguration
+from codeflare_sdk.cluster.auth import TokenAuthentication
+
+
+def test_sdk_functionality():
+    pass
```
Comment (Maxusmusti) [inline tests/resources/test_mcad_ray.py:7]: missing newline at EOF

File: tests/basictests/distributed-workloads.sh [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+source ${MY_DIR}/../util
+
+os::test::junit::declare_suite_start "$MY_SCRIPT"
+
+function install_codeflare_operator() {
+    haeder: "Installing Codeflare Operator"
+}
+
+function install_distributred_workloads_kfdef(){
```
Comment (Maxusmusti) [inline tests/basictests/distributed-workloads.sh:17]: distributred -> distributed

**Review by Maxusmusti — CHANGES_REQUESTED** [review]
Some very minor nits, once fixed it LGTM

File: tests/basictests/distributed-workloads.sh [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+MY_DIR=$(readlink -f `dirname "${BASH_SOURCE[0]}"`)
+
+RESOURCEDIR="${MY_DIR}/../resources"
+
+source ${MY_DIR}/../util
+
+os::test::junit::declare_suite_start "$MY_SCRIPT"
+
+function install_codeflare_operator() {
+    haeder: "Installing Codeflare Operator"
```
Comment (anishasthana) [inline tests/basictests/distributed-workloads.sh:14]:
```suggestion
    header: "Installing CodeFlare Operator"
```

**Review by anishasthana — CHANGES_REQUESTED** [review]

**Comment by MichaelClifford** [comment]
:upside_down_face: Sorry for the spelling errors. Should be fixed now.

**Review by Maxusmusti — APPROVED** [review]
LGTM

**Comment by tedhtchang** [comment]
which function will test the Notebook ?

**Comment by MichaelClifford** [comment]
We might need an additional tests for the notebook image itself.  However, I'm assuming it will be part of both `tests_mcad_ray_functionality()` and `tests_mcad_torchx_functionality()` as those will run the SDK from the notebook image.

File: tests/basictests/distributed-workloads.sh [hunk 5]
```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+source $TEST_DIR/common
```
Comment (KPostOffice) [inline tests/basictests/distributed-workloads.sh:3]: Do you know where this actually comes from? Is this included in the image or something. I saw it in the odh-manifests too. I assume it provides the functions used for testing like `os::cmd::expect_success`

**Review by KPostOffice — COMMENTED** [review]
This looks good to me. Not blocking, just a question.

File: tests/basictests/distributed-workloads.sh (hunk 5 above)
Comment (MichaelClifford) [reply tests/basictests/distributed-workloads.sh:3]: I think you are right, @KPostOffice. you can see more about the testing framework here: https://github.com/opendatahub-io/odh-manifests/blob/master/tests/TESTING.md

File: tests/basictests/distributed-workloads.sh [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    header: "Uninstalling distributed workloads kfdef"
+}
+
+function uninstall_codeflare_operator() {
+    header: "Uninstalling Codeflare Operator"
+}
+
+
+install_codeflare_operator
+install_distributred_workloads_kfdef
```
Comment (KPostOffice) [inline tests/basictests/distributed-workloads.sh:50]:
```suggestion
install_distributed_workloads_kfdef
```

**Comment by MichaelClifford** [comment]
@anishasthana @KPostOffice @tedhtchang @Maxusmusti 
I added a test README along with util file needed for testing and simplified the example tests. 

Please take another look and let me know if its good to go :smile:

**Comment by jbusche** [comment]
LGTM, thanks Michael!

**Review by anishasthana — APPROVED** [review]

**Pull request merged**